--- work-old/smalloc.c	Wed Jan 13 15:16:17 2021
+++ work/smalloc.c	Wed Jan 13 15:16:51 2021
@@ -241,7 +241,7 @@
 	 * This will also remove the temporary file we used as a backing
 	 * store, it was already unlinked
 	 */
-	munmap(pool->map, pool->mmap_size);
+	munmap((char *)pool->map, pool->mmap_size);
 
 	if (pool->lock)
 		fio_sem_remove(pool->lock);
@@ -254,7 +254,7 @@
 	for (i = 0; i < nr_pools; i++)
 		cleanup_pool(&mp[i]);
 
-	munmap(mp, MAX_POOLS * sizeof(struct pool));
+	munmap((char *)mp, MAX_POOLS * sizeof(struct pool));
 }
 
 #ifdef SMALLOC_REDZONE
