diff -ur cockroach-v20.1.5-pristine/src/github.com/cockroachdb/cockroach/c-deps/rocksdb/monitoring/perf_context.cc cockroach-v20.1.5/src/github.com/cockroachdb/cockroach/c-deps/rocksdb/monitoring/perf_context.cc
--- cockroach-v20.1.5-pristine/src/github.com/cockroachdb/cockroach/c-deps/rocksdb/monitoring/perf_context.cc	Mon Aug 24 12:36:13 2020
+++ cockroach-v20.1.5/src/github.com/cockroachdb/cockroach/c-deps/rocksdb/monitoring/perf_context.cc	Thu Sep  3 16:51:25 2020
@@ -12,27 +12,19 @@
 #if defined(NPERF_CONTEXT) || !defined(ROCKSDB_SUPPORT_THREAD_LOCAL)
 PerfContext perf_context;
 #else
-#if defined(OS_SOLARIS)
-__thread PerfContext perf_context_;
-#else
 thread_local PerfContext perf_context;
 #endif
-#endif
 
 PerfContext* get_perf_context() {
 #if defined(NPERF_CONTEXT) || !defined(ROCKSDB_SUPPORT_THREAD_LOCAL)
   return &perf_context;
 #else
-#if defined(OS_SOLARIS)
-  return &perf_context_;
-#else
   return &perf_context;
 #endif
-#endif
 }
 
 PerfContext::~PerfContext() {
-#if !defined(NPERF_CONTEXT) && defined(ROCKSDB_SUPPORT_THREAD_LOCAL) && !defined(OS_SOLARIS)
+#if !defined(NPERF_CONTEXT) && defined(ROCKSDB_SUPPORT_THREAD_LOCAL)
   ClearPerLevelPerfContext();
 #endif
 }
diff -ur cockroach-v20.1.5-pristine/src/github.com/cockroachdb/cockroach/c-deps/rocksdb/monitoring/perf_context_imp.h cockroach-v20.1.5/src/github.com/cockroachdb/cockroach/c-deps/rocksdb/monitoring/perf_context_imp.h
--- cockroach-v20.1.5-pristine/src/github.com/cockroachdb/cockroach/c-deps/rocksdb/monitoring/perf_context_imp.h	Mon Aug 24 12:36:13 2020
+++ cockroach-v20.1.5/src/github.com/cockroachdb/cockroach/c-deps/rocksdb/monitoring/perf_context_imp.h	Thu Sep  3 16:51:25 2020
@@ -12,13 +12,8 @@
 #if defined(NPERF_CONTEXT) || !defined(ROCKSDB_SUPPORT_THREAD_LOCAL)
 extern PerfContext perf_context;
 #else
-#if defined(OS_SOLARIS)
-extern __thread PerfContext perf_context_;
-#define perf_context (*get_perf_context())
-#else
 extern thread_local PerfContext perf_context;
 #endif
-#endif
 
 #if defined(NPERF_CONTEXT)
 
