diff -ur cockroach-v20.1.5-pristine/src/github.com/cockroachdb/cockroach/pkg/util/sysutil/sysutil_unix.go cockroach-v20.1.5/src/github.com/cockroachdb/cockroach/pkg/util/sysutil/sysutil_unix.go
--- cockroach-v20.1.5-pristine/src/github.com/cockroachdb/cockroach/pkg/util/sysutil/sysutil_unix.go	Mon Aug 24 12:36:10 2020
+++ cockroach-v20.1.5/src/github.com/cockroachdb/cockroach/pkg/util/sysutil/sysutil_unix.go	Thu Sep  3 18:23:42 2020
@@ -8,7 +8,7 @@
 // by the Apache License, Version 2.0, included in the file
 // licenses/APL.txt.
 
-// +build !windows
+// +build !windows,!illumos
 
 //lint:file-ignore Unconvert (redundant conversions are necessary for cross-platform compatibility)
 
diff -ur cockroach-v20.1.5-pristine/src/github.com/cockroachdb/cockroach/pkg/util/sysutil/sysutil_unix_test.go cockroach-v20.1.5/src/github.com/cockroachdb/cockroach/pkg/util/sysutil/sysutil_unix_test.go
--- cockroach-v20.1.5-pristine/src/github.com/cockroachdb/cockroach/pkg/util/sysutil/sysutil_unix_test.go	Mon Aug 24 12:36:10 2020
+++ cockroach-v20.1.5/src/github.com/cockroachdb/cockroach/pkg/util/sysutil/sysutil_unix_test.go	Thu Sep  3 18:23:36 2020
@@ -8,7 +8,7 @@
 // by the Apache License, Version 2.0, included in the file
 // licenses/APL.txt.
 
-// +build !windows
+// +build !windows,!illumos
 
 package sysutil
 
