diff --git a/src/github.com/cockroachdb/cockroach/pkg/util/log/stderr_redirect_unix.go b/src/github.com/cockroachdb/cockroach/pkg/util/log/stderr_redirect_unix.go
--- a/src/github.com/cockroachdb/cockroach/pkg/util/log/stderr_redirect_unix.go
+++ b/src/github.com/cockroachdb/cockroach/pkg/util/log/stderr_redirect_unix.go
@@ -8,7 +8,7 @@
 // by the Apache License, Version 2.0, included in the file
 // licenses/APL.txt.
 
-// +build !windows
+// +build !windows,!illumos
 
 package log
 
