diff -ur cockroach-{pristine,patched}/src/github.com/cockroachdb/cockroach/pkg/util/log/stderr_redirect_unix.go
--- cockroach-pristine/src/github.com/cockroachdb/cockroach/pkg/util/log/stderr_redirect_unix.go        Wed Aug 17 10:40:01 2022
+++ cockroach-patched/src/github.com/cockroachdb/cockroach/pkg/util/log/stderr_redirect_unix.go Wed Aug 17 11:08:45 2022
@@ -8,8 +8,8 @@
 // by the Apache License, Version 2.0, included in the file
 // licenses/APL.txt.
 
-//go:build !windows
-// +build !windows
+//go:build !windows && !illumos
+// +build !windows,!illumos
 
 package log
 
