diff -ur cockroach-v20.1.5-pristine/src/github.com/cockroachdb/cockroach/pkg/cli/start_jemalloc.go cockroach-v20.1.5/src/github.com/cockroachdb/cockroach/pkg/cli/start_jemalloc.go
--- cockroach-v20.1.5-pristine/src/github.com/cockroachdb/cockroach/pkg/cli/start_jemalloc.go	Mon Aug 24 12:36:09 2020
+++ cockroach-v20.1.5/src/github.com/cockroachdb/cockroach/pkg/cli/start_jemalloc.go	Fri Sep  4 00:13:50 2020
@@ -14,7 +14,7 @@
 
 // #cgo CPPFLAGS: -DJEMALLOC_NO_DEMANGLE
 // #cgo LDFLAGS: -ljemalloc
-// #cgo dragonfly freebsd LDFLAGS: -lm
+// #cgo illumos dragonfly freebsd LDFLAGS: -lm
 // #cgo linux LDFLAGS: -lrt -lm -lpthread
 //
 // #include <jemalloc/jemalloc.h>
diff -ur cockroach-v20.1.5-pristine/src/github.com/cockroachdb/cockroach/pkg/server/status/runtime_jemalloc.go cockroach-v20.1.5/src/github.com/cockroachdb/cockroach/pkg/server/status/runtime_jemalloc.go
--- cockroach-v20.1.5-pristine/src/github.com/cockroachdb/cockroach/pkg/server/status/runtime_jemalloc.go	Mon Aug 24 12:36:09 2020
+++ cockroach-v20.1.5/src/github.com/cockroachdb/cockroach/pkg/server/status/runtime_jemalloc.go	Fri Sep  4 00:14:08 2020
@@ -14,7 +14,7 @@
 
 // #cgo CPPFLAGS: -DJEMALLOC_NO_DEMANGLE
 // #cgo LDFLAGS: -ljemalloc
-// #cgo dragonfly freebsd LDFLAGS: -lm
+// #cgo illumos dragonfly freebsd LDFLAGS: -lm
 // #cgo linux LDFLAGS: -lrt -lm -lpthread
 //
 // #include <jemalloc/jemalloc.h>
