--- a/src/github.com/cockroachdb/cockroach/Makefile	Mon Nov  2 07:10:59 2020
+++ b/src/github.com/cockroachdb/cockroach/Makefile	Wed Nov 11 11:32:05 2020
@@ -176,13 +176,6 @@
 # Avoid reusing GOFLAGS as that is overwritten by various release processes.
 GOMODVENDORFLAGS := -mod=vendor
 
-ifeq "$(findstring -j,$(shell ps -o args= $$PPID))" ""
-ifdef NCPUS
-MAKEFLAGS += -j$(NCPUS)
-$(info Running make with -j$(NCPUS))
-endif
-endif
-
 help: ## Print help for targets with comments.
 	@echo "Usage:"
 	@echo "  make [target...] [VAR=foo VAR2=bar...]"
@@ -260,7 +253,7 @@
 
 # We install our vendored tools to a directory within this repository to avoid
 # overwriting any user-installed binaries of the same name in the default GOBIN.
-GO_INSTALL := GOBIN='$(abspath bin)' GOFLAGS= $(GO) install
+GO_INSTALL := GOBIN='$(abspath bin)' GOFLAGS= $(GO) install $(GOMODVENDORFLAGS)
 
 # Prefer tools we've installed with go install and Yarn to those elsewhere on
 # the PATH.
@@ -725,7 +718,7 @@
 	@uptodate $(GEOS_NATIVE_LIB_DIR)/libgeos.$(DYN_EXT) $(GEOS_SRC_DIR) || $(MAKE) --no-print-directory -C $(GEOS_DIR) geos_c
 	mkdir -p $(DYN_LIB_DIR)
 	rm -f $(DYN_LIB_DIR)/lib{geos,geos_c}.$(DYN_EXT)
-	cp -L $(GEOS_NATIVE_LIB_DIR)/lib{geos,geos_c}.$(DYN_EXT) $(DYN_LIB_DIR)
+	cp $(GEOS_NATIVE_LIB_DIR)/lib{geos,geos_c}.$(DYN_EXT) $(DYN_LIB_DIR)
 
 $(LIBPROJ): $(PROJ_DIR)/Makefile bin/uptodate .ALWAYS_REBUILD
 	@uptodate $@ $(PROJ_SRC_DIR) || $(MAKE) --no-print-directory -C $(PROJ_DIR) proj
@@ -1532,8 +1525,6 @@
 	--protoc-flags "-Ipkg -I$(GOGO_PROTOBUF_PATH) -I$(COREOS_PATH) -I$(GRPC_GATEWAY_GOOGLEAPIS_PATH) -I$(ERRORS_PATH) -I$(PROMETHEUS_PATH) ./pkg/server/serverpb/status.proto ./pkg/server/serverpb/admin.proto ./pkg/server/status/statuspb/status.proto"
 	touch $@
 
-.PHONY: docs/generated/redact_safe.md
-
 docs/generated/redact_safe.md:
 	./build/bazelutil/generate_redact_safe.sh >$@.tmp || { rm -f $@.tmp; exit 1; }
 	@mv -f $@.tmp $@
