--- cockroach-v20.2.0-rc.4/src/github.com/cockroachdb/cockroach/pkg/ui/src/routes/visualization.tsx	Mon Nov  2 07:11:00 2020
+++ cockroach-v20.2.0-rc.4-patched/src/github.com/cockroachdb/cockroach/pkg/ui/src/routes/visualization.tsx	Wed Nov 11 11:13:35 2020
@@ -31,11 +31,13 @@
 
 export default function createClusterOverviewRoutes(): JSX.Element {
   return (
-    <Route path="overview" component={ ClusterOverview } >
-      <Switch>
-        <Route exact path="/" component={() => <Redirect to="list" />}/>
-        <Route path="list" component={ NodesWrapper } />
-      </Switch>
+    <Route path="/overview">
+      <ClusterOverview>
+        <Switch>
+          <Redirect exact from="/overview" to="/overview/list" />
+          <Route path="/overview/list" component={ NodesWrapper } />
+        </Switch>
+      </ClusterOverview>
     </Route>
   );
 }
