diff -ru cockroach-v21.1.0-pristine/src/github.com/cockroachdb/cockroach/c-deps/libroach/stack_trace.cc cockroach-v21.1.0/src/github.com/cockroachdb/cockroach/c-deps/libroach/stack_trace.cc
--- cockroach-v21.1.0-pristine/src/github.com/cockroachdb/cockroach/c-deps/libroach/stack_trace.cc	Mon May 17 06:41:15 2021
+++ cockroach-v21.1.0/src/github.com/cockroachdb/cockroach/c-deps/libroach/stack_trace.cc	Tue May 18 14:13:16 2021
@@ -329,6 +329,7 @@
 #else  // !defined(OS_LINUX) || !defined(__GLIBC__)
 
 #include <stdlib.h>
+#include <cstring>
 
 std::string DumpThreadStacks() {
   return "thread stacks only available on Linux/Glibc";
