diff --git a/contrib/abseil-cpp/absl/copts/AbseilConfigureCopts.cmake b/contrib/abseil-cpp/absl/copts/AbseilConfigureCopts.cmake
index 77d4ace..30891a4 100644
--- a/contrib/abseil-cpp/absl/copts/AbseilConfigureCopts.cmake
+++ b/contrib/abseil-cpp/absl/copts/AbseilConfigureCopts.cmake
@@ -72,3 +72,5 @@ elseif(NOT "${CMAKE_CXX_STANDARD}")
 else()
   set(ABSL_CXX_STANDARD "${CMAKE_CXX_STANDARD}")
 endif()
+# Absl cannot work with C++20, which removed some typedefs for std::allocator
+set(ABSL_CXX_STANDARD 17)
