From cf79e61bd947baa46672c78adc799dd676d47c49 Mon Sep 17 00:00:00 2001
From: Oxide Computer Company <eng@oxide.computer>
Date: Fri, 24 Oct 2025 15:23:44 -0400
Subject: [PATCH 03/61] Disable thread safety analysis

---
 cmake/warnings.cmake | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/cmake/warnings.cmake b/cmake/warnings.cmake
index 807d92d9..bf9900ad 100644
--- a/cmake/warnings.cmake
+++ b/cmake/warnings.cmake
@@ -43,6 +43,8 @@ no_warning(vla)
 no_warning(weak-template-vtables)
 no_warning(weak-vtables)
 no_warning(thread-safety-negative) # experimental flag, too many false positives
+no_warning(thread-safety-analysis) # experimental flag, too many false positives
+no_warning(thread-safety-attributes) # experimental flag, too many false positives
 no_warning(enum-constexpr-conversion) # breaks magic-enum library in clang-16
 no_warning(unsafe-buffer-usage) # too aggressive
 no_warning(switch-default) # conflicts with "defaults in a switch covering all enum values"
-- 
2.50.1 (Apple Git-155)

