From b08418d5406972eb81088dfdb8438f15a465e520 Mon Sep 17 00:00:00 2001
From: Oxide Computer Company <eng@oxide.computer>
Date: Wed, 22 Nov 2023 11:45:44 +0000
Subject: [PATCH 11/61] Enable SSE2 divide implementation on illumos

---
 src/Functions/divide/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Functions/divide/CMakeLists.txt b/src/Functions/divide/CMakeLists.txt
index 03c2faaf3c..bec9d3c66c 100644
--- a/src/Functions/divide/CMakeLists.txt
+++ b/src/Functions/divide/CMakeLists.txt
@@ -1,6 +1,6 @@
 # A library for integer division by constant with CPU dispatching.
 
-if (ARCH_AMD64)
+if (ARCH_AMD64 OR OS_SUNOS)
     add_library(divide_impl_sse2 divideImpl.cpp)
     target_compile_options(divide_impl_sse2 PRIVATE -msse2 -DNAMESPACE=SSE2)
     target_link_libraries(divide_impl_sse2 ch_contrib::libdivide)
-- 
2.40.1

