From 53bbb150e618551996264b30872586ec102f668e Mon Sep 17 00:00:00 2001
From: Oxide Computer Company <eng@oxide.computer>
Date: Wed, 22 Nov 2023 12:39:46 +0000
Subject: [PATCH 16/61] illumos support for pollPid

---
 src/Common/waitForPid.cpp | 22 ++++++++++++++++++++++
 1 file changed, 22 insertions(+)

diff --git a/src/Common/waitForPid.cpp b/src/Common/waitForPid.cpp
index 0ec1081135..539ec7a819 100644
--- a/src/Common/waitForPid.cpp
+++ b/src/Common/waitForPid.cpp
@@ -153,6 +153,28 @@ static PollPidResult pollPid(pid_t pid, int timeout_in_ms)
 
     return result;
 }
+#elif defined(__sun)
+#include <libproc.h>
+
+namespace DB
+{
+
+static PollPidResult pollPid(pid_t pid, int timeout_in_ms)
+{
+    PollPidResult result = PollPidResult::FAILED;
+    int rc, perr;
+    struct ps_prochandle *hdl;
+    hdl = Pgrab(pid, PGRAB_RETAIN | PGRAB_FORCE | PGRAB_NOSTOP, &perr);
+    if (hdl == NULL)
+        return PollPidResult::FAILED;
+    rc = Pstopstatus(hdl, PCWSTOP, timeout_in_ms);
+    if (rc < 0 && errno == ENOENT)
+        result = PollPidResult::RESTART;
+    if (rc == 0 && Pstate(hdl) != PS_RUN)
+        result = PollPidResult::RESTART;
+    Pfree(hdl);
+    return result;
+}
 #else
     #error "Unsupported OS type"
 #endif
-- 
2.40.1

