From 23bbd0f7cf56f4b46703f2d4cbba93e36cfe174e Mon Sep 17 00:00:00 2001
From: Oxide Computer Company <eng@oxide.computer>
Date: Fri, 24 Oct 2025 16:54:47 -0400
Subject: [PATCH 32/61] Loggers: unresolved std::time_put

---
 src/Loggers/Loggers.cpp | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/src/Loggers/Loggers.cpp b/src/Loggers/Loggers.cpp
index 35b96bce..67dd35bb 100644
--- a/src/Loggers/Loggers.cpp
+++ b/src/Loggers/Loggers.cpp
@@ -49,7 +49,14 @@ static std::string renderFileNameTemplate(time_t now, const std::string & file_p
     std::tm buf;
     localtime_r(&now, &buf);
     std::ostringstream ss; // STYLE_CHECK_ALLOW_STD_STRING_STREAM
-    ss << std::put_time(&buf, path.filename().c_str());
+    #ifdef __illumos__
+        char timestr[MAXPATHLEN];
+
+        strftime(timestr, sizeof (timestr), path.filename().c_str(), &buf);
+        ss << timestr;
+    #else
+        ss << std::put_time(&buf, path.filename().c_str());
+    #endif
     return path.replace_filename(ss.str());
 }
 
-- 
2.50.1 (Apple Git-155)

