From 39151dc3aec237c4360eb4b5b757bafdba286bcd Mon Sep 17 00:00:00 2001
From: Oxide Computer Company <eng@oxide.computer>
Date: Thu, 23 Nov 2023 09:31:06 +0000
Subject: [PATCH 45/61] contrib/curl: Add illumos support

---
 contrib/curl-cmake/curl_config.h | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/contrib/curl-cmake/curl_config.h b/contrib/curl-cmake/curl_config.h
index f56ba3eccd..1aaa0bc341 100644
--- a/contrib/curl-cmake/curl_config.h
+++ b/contrib/curl-cmake/curl_config.h
@@ -50,3 +50,9 @@
 #define ENABLE_IPV6
 #define USE_OPENSSL
 #define USE_THREADS_POSIX
+
+#ifdef __illumos__
+#define HAVE_POSIX_STRERROR_R 1
+#define HAVE_STRERROR_R 1
+#endif
+
-- 
2.40.1

