From 197ada16cc7942ec5b2457d0056790f5c4da4845 Mon Sep 17 00:00:00 2001
From: Alex Wilson <alex@cooperi.net>
Date: Mon, 29 May 2023 09:58:55 +1000
Subject: [PATCH 01/42] 02-poco-cmake.patch

diff -wpruN --no-dereference '--exclude=*.orig' a~/base/poco/Net/CMakeLists.txt a/base/poco/Net/CMakeLists.txt
--- a~/base/poco/Net/CMakeLists.txt	1970-01-01 00:00:00
+++ a/base/poco/Net/CMakeLists.txt	1970-01-01 00:00:00
@@ -9,6 +9,10 @@ elseif (OS_DARWIN OR OS_FREEBSD)
     target_compile_definitions (_poco_net PUBLIC POCO_HAVE_FD_POLL)
 endif ()
 
+if (OS_SUNOS)
+    target_link_libraries (_poco_net PUBLIC socket nsl)
+endif ()
+
 # TODO: remove these warning exclusions
 target_compile_options (_poco_net
     PRIVATE
