From 0fe9bef938fc401432bc9b3f9073e6fbd69fba51 Mon Sep 17 00:00:00 2001
From: Alex Wilson <alex@cooperi.net>
Date: Mon, 29 May 2023 09:58:57 +1000
Subject: [PATCH 07/42] 08-PerformanceAdaptors.patch

diff -wpruN --no-dereference '--exclude=*.orig' a~/src/Functions/PerformanceAdaptors.h a/src/Functions/PerformanceAdaptors.h
--- a~/src/Functions/PerformanceAdaptors.h	1970-01-01 00:00:00
+++ a/src/Functions/PerformanceAdaptors.h	1970-01-01 00:00:00
@@ -99,7 +99,7 @@ namespace detail
             ///  when there is no statistical significant difference between them.
             double sigma() const
             {
-                return mean() / sqrt(adjustedCount());
+                return mean() / sqrt(static_cast<double>(adjustedCount()));
             }
 
             void run()
