From f6afea679eb87af3fea7ace36312300ab8f7d021 Mon Sep 17 00:00:00 2001
From: Alex Wilson <alex@cooperi.net>
Date: Mon, 29 May 2023 09:58:58 +1000
Subject: [PATCH 08/42] 09-QuantileExact.patch

diff -wpruN --no-dereference '--exclude=*.orig' a~/src/AggregateFunctions/QuantileExact.h a/src/AggregateFunctions/QuantileExact.h
--- a~/src/AggregateFunctions/QuantileExact.h	1970-01-01 00:00:00
+++ a/src/AggregateFunctions/QuantileExact.h	1970-01-01 00:00:00
@@ -275,11 +275,11 @@ struct QuantileExactLow : public Quantil
                 auto s = array.size();
                 if (s % 2 == 1)
                 {
-                    n = static_cast<size_t>(floor(s / 2));
+                    n = static_cast<size_t>(static_cast<double>(floor(static_cast<double>(s / 2))));
                 }
                 else
                 {
-                    n = static_cast<size_t>((floor(s / 2)) - 1);
+                    n = static_cast<size_t>((static_cast<double>(floor(static_cast<double>(s / 2)))) - 1);
                 }
             }
             else
@@ -311,11 +311,11 @@ struct QuantileExactLow : public Quantil
                     auto s = array.size();
                     if (s % 2 == 1)
                     {
-                        n = static_cast<size_t>(floor(s / 2));
+                        n = static_cast<size_t>(static_cast<double>(floor(static_cast<double>(s / 2))));
                     }
                     else
                     {
-                        n = static_cast<size_t>(floor((s / 2) - 1));
+                        n = static_cast<size_t>(static_cast<double>(floor(static_cast<double>(s / 2) - 1)));
                     }
                 }
                 else
@@ -355,7 +355,7 @@ struct QuantileExactHigh : public Quanti
             if (level == 0.5)
             {
                 auto s = array.size();
-                n = static_cast<size_t>(floor(s / 2));
+                n = static_cast<size_t>(static_cast<double>(floor(static_cast<double>(s / 2))));
             }
             else
             {
@@ -383,7 +383,7 @@ struct QuantileExactHigh : public Quanti
                 if (level == 0.5)
                 {
                     auto s = array.size();
-                    n = static_cast<size_t>(floor(s / 2));
+                    n = static_cast<size_t>(static_cast<double>(floor(static_cast<double>(s / 2))));
                 }
                 else
                 {
