From f6afea679eb87af3fea7ace36312300ab8f7d021 Mon Sep 17 00:00:00 2001
From: Alex Wilson <alex@cooperi.net>
Date: Mon, 29 May 2023 09:58:58 +1000
Subject: [PATCH 08/42] 09-QuantileExact.patch

---
 src/AggregateFunctions/QuantileExact.h | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/src/AggregateFunctions/QuantileExact.h b/src/AggregateFunctions/QuantileExact.h
index 9be24689d1..118499b158 100644
--- a/src/AggregateFunctions/QuantileExact.h
+++ b/src/AggregateFunctions/QuantileExact.h
@@ -270,11 +270,11 @@ struct QuantileExactLow : public QuantileExactBase<Value, QuantileExactLow<Value
                 auto s = array.size();
                 if (s % 2 == 1)
                 {
-                    n = static_cast<size_t>(floor(s / 2));
+                    n = static_cast<size_t>(static_cast<double>(floor(static_cast<double>(s / 2))));
                 }
                 else
                 {
-                    n = static_cast<size_t>((floor(s / 2)) - 1);
+                    n = static_cast<size_t>((static_cast<double>(floor(static_cast<double>(s / 2)))) - 1);
                 }
             }
             else
@@ -306,11 +306,11 @@ struct QuantileExactLow : public QuantileExactBase<Value, QuantileExactLow<Value
                     auto s = array.size();
                     if (s % 2 == 1)
                     {
-                        n = static_cast<size_t>(floor(s / 2));
+                        n = static_cast<size_t>(static_cast<double>(floor(static_cast<double>(s / 2))));
                     }
                     else
                     {
-                        n = static_cast<size_t>(floor((s / 2) - 1));
+                        n = static_cast<size_t>(static_cast<double>(floor(static_cast<double>(s / 2) - 1)));
                     }
                 }
                 else
@@ -350,7 +350,7 @@ struct QuantileExactHigh : public QuantileExactBase<Value, QuantileExactHigh<Val
             if (level == 0.5)
             {
                 auto s = array.size();
-                n = static_cast<size_t>(floor(s / 2));
+                n = static_cast<size_t>(static_cast<double>(floor(static_cast<double>(s / 2))));
             }
             else
             {
@@ -378,7 +378,7 @@ struct QuantileExactHigh : public QuantileExactBase<Value, QuantileExactHigh<Val
                 if (level == 0.5)
                 {
                     auto s = array.size();
-                    n = static_cast<size_t>(floor(s / 2));
+                    n = static_cast<size_t>(static_cast<double>(floor(static_cast<double>(s / 2))));
                 }
                 else
                 {
-- 
2.40.1

