From 8d4e510d1d145ff9b6425ebe308bd701616e5e07 Mon Sep 17 00:00:00 2001
From: Alex Wilson <alex@cooperi.net>
Date: Mon, 29 May 2023 09:58:58 +1000
Subject: [PATCH 09/42] 10-SimpleMergeSelector.patch

---
 src/Storages/MergeTree/SimpleMergeSelector.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/Storages/MergeTree/SimpleMergeSelector.cpp b/src/Storages/MergeTree/SimpleMergeSelector.cpp
index 434d44022d..064e4c12d1 100644
--- a/src/Storages/MergeTree/SimpleMergeSelector.cpp
+++ b/src/Storages/MergeTree/SimpleMergeSelector.cpp
@@ -207,8 +207,8 @@ SimpleMergeSelector::PartsRange SimpleMergeSelector::select(
     Estimator estimator;
 
     /// Precompute logarithm of settings boundaries, because log function is quite expensive in terms of performance
-    const double min_size_to_lower_base_log = log(1 + settings.min_size_to_lower_base);
-    const double max_size_to_lower_base_log = log(1 + settings.max_size_to_lower_base);
+    const double min_size_to_lower_base_log = log(static_cast<double>(1 + settings.min_size_to_lower_base));
+    const double max_size_to_lower_base_log = log(static_cast<double>(1 + settings.max_size_to_lower_base));
 
     for (const auto & part_range : parts_ranges)
         selectWithinPartition(part_range, max_total_size_to_merge, estimator, settings, min_size_to_lower_base_log, max_size_to_lower_base_log);
-- 
2.40.1

