From 2680da59519c304341040dc4ab15dde4ab5d4a1d Mon Sep 17 00:00:00 2001
From: Alex Wilson <alex@cooperi.net>
Date: Mon, 29 May 2023 09:58:59 +1000
Subject: [PATCH 12/42] 13-s2geometry-endian.patch

---
 contrib/s2geometry/src/s2/base/port.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/contrib/s2geometry/src/s2/base/port.h b/contrib/s2geometry/src/s2/base/port.h
index 0efaba8424..f814891219 100644
--- a/contrib/s2geometry/src/s2/base/port.h
+++ b/contrib/s2geometry/src/s2/base/port.h
@@ -149,7 +149,7 @@
 // -----------------------------------------------------------------------------
 
 // IS_LITTLE_ENDIAN, IS_BIG_ENDIAN
-#if defined __linux__ || defined OS_ANDROID || defined(__ANDROID__)
+#if defined __linux__ || defined OS_ANDROID || defined(__ANDROID__) || defined(OS_SUNOS)
 // TODO(user): http://b/21460321; use one of OS_ANDROID or __ANDROID__.
 // _BIG_ENDIAN
 #include <endian.h>
-- 
2.40.1

