--- a/contrib/rocksdb/monitoring/perf_context.cc
+++ b/contrib/rocksdb/monitoring/perf_context.cc
@@ -14,11 +14,7 @@
 // Put here just to make get_perf_context() simple without ifdef.
 PerfContext perf_context;
 #elif defined(ROCKSDB_SUPPORT_THREAD_LOCAL)
-#if defined(OS_SOLARIS)
-__thread PerfContext perf_context;
-#else   // OS_SOLARIS
 thread_local PerfContext perf_context;
-#endif  // OS_SOLARIS
 #else
 #error "No thread-local support. Disable perf context with -DNPERF_CONTEXT."
 #endif
--- a/contrib/rocksdb/env/fs_posix.cc
+++ b/contrib/rocksdb/env/fs_posix.cc
@@ -635,7 +635,9 @@
       // filter out '.' and '..' directory entries
       // which appear only on some platforms
       const bool ignore =
+#ifndef OS_SUNOS
           entry->d_type == DT_DIR &&
+#endif
           (strcmp(entry->d_name, ".") == 0 || strcmp(entry->d_name, "..") == 0);
       if (!ignore) {
         result->push_back(entry->d_name);
