From 448a2e8d2a4457a163075ae901dbdb5ba43f5c81 Mon Sep 17 00:00:00 2001
From: Alex Wilson <alex@cooperi.net>
Date: Mon, 29 May 2023 09:59:01 +1000
Subject: [PATCH 16/42] 17-IBridge.patch

---
 src/Bridge/IBridge.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/Bridge/IBridge.cpp b/src/Bridge/IBridge.cpp
index d30e0a4c85..e6f032f98a 100644
--- a/src/Bridge/IBridge.cpp
+++ b/src/Bridge/IBridge.cpp
@@ -172,6 +172,7 @@ void IBridge::initialize(Application & self)
     max_server_connections = config().getUInt("max-server-connections", 1024);
     keep_alive_timeout = config().getUInt64("keep-alive-timeout", 10);
 
+#ifndef OS_SUNOS
     struct rlimit limit;
     const UInt64 gb = 1024 * 1024 * 1024;
 
@@ -196,6 +197,7 @@ void IBridge::initialize(Application & self)
     {
         LOG_WARNING(log, "Failed to set OOM score, error: {}", e.what());
     }
+#endif /* !__illumos__ */
 
     initializeTerminationAndSignalProcessing();
 
-- 
2.40.1

