--- a/base/bridge/IBridge.cpp
+++ b/base/bridge/IBridge.cpp
@@ -171,6 +171,7 @@
     max_server_connections = config().getUInt("max-server-connections", 1024);
     keep_alive_timeout = config().getUInt64("keep-alive-timeout", 10);
 
+#ifndef OS_SUNOS
     struct rlimit limit;
     const UInt64 gb = 1024 * 1024 * 1024;
 
@@ -195,6 +196,7 @@
     {
         LOG_WARNING(log, "Failed to set OOM score, error: {}", e.what());
     }
+#endif /* !__illumos__ */
 
     initializeTerminationAndSignalProcessing();
 
