From 4563253b81f57e4ff43bb87307ac6e64b0b29e06 Mon Sep 17 00:00:00 2001
From: Alex Wilson <alex@cooperi.net>
Date: Mon, 29 May 2023 09:59:01 +1000
Subject: [PATCH 17/42] 19-Keepalive.patch

diff -wpruN --no-dereference '--exclude=*.orig' a~/src/Client/Connection.cpp a/src/Client/Connection.cpp
--- a~/src/Client/Connection.cpp	1970-01-01 00:00:00
+++ a/src/Client/Connection.cpp	1970-01-01 00:00:00
@@ -171,13 +171,21 @@ void Connection::connect(const Connectio
         if (tcp_keep_alive_timeout_in_sec)
         {
             socket->setKeepAlive(true);
-            socket->setOption(IPPROTO_TCP,
-#if defined(TCP_KEEPALIVE)
-                TCP_KEEPALIVE
-#else
-                TCP_KEEPIDLE  // __APPLE__
+#if defined(TCP_KEEPALIVE_THRESHOLD)
+            int thresh = timeouts.tcp_keep_alive_timeout.totalMilliseconds() / 2;
+            if (thresh < 5000)
+                thresh = 5000;
+            int intvl = timeouts.tcp_keep_alive_timeout.totalSeconds() / 6;
+            if (intvl < 2)
+                intvl = 2;
+            socket->setOption(IPPROTO_TCP, TCP_KEEPALIVE_THRESHOLD, thresh);
+            socket->setOption(IPPROTO_TCP, TCP_KEEPINTVL, intvl);
+            socket->setOption(IPPROTO_TCP, TCP_KEEPCNT, 3);
+#elif defined(TCP_KEEPALIVE)
+            socket->setOption(IPPROTO_TCP, TCP_KEEPALIVE, tcp_keep_alive_timeout_in_sec);
+#elif defined(TCP_KEEPIDLE)
+            socket->setOption(IPPROTO_TCP, TCP_KEEPIDLE, tcp_keep_alive_timeout_in_sec);
 #endif
-                , tcp_keep_alive_timeout_in_sec);
         }
 
         in = std::make_shared<ReadBufferFromPocoSocket>(*socket);
