From 4563253b81f57e4ff43bb87307ac6e64b0b29e06 Mon Sep 17 00:00:00 2001
From: Alex Wilson <alex@cooperi.net>
Date: Mon, 29 May 2023 09:59:01 +1000
Subject: [PATCH 17/42] 19-Keepalive.patch

---
 src/Client/Connection.cpp | 20 ++++++++++++++------
 1 file changed, 14 insertions(+), 6 deletions(-)

diff --git a/src/Client/Connection.cpp b/src/Client/Connection.cpp
index 5047f958a1..15a69f6894 100644
--- a/src/Client/Connection.cpp
+++ b/src/Client/Connection.cpp
@@ -150,13 +150,21 @@ void Connection::connect(const ConnectionTimeouts & timeouts)
         if (timeouts.tcp_keep_alive_timeout.totalSeconds())
         {
             socket->setKeepAlive(true);
-            socket->setOption(IPPROTO_TCP,
-#if defined(TCP_KEEPALIVE)
-                TCP_KEEPALIVE
-#else
-                TCP_KEEPIDLE  // __APPLE__
+#if defined(TCP_KEEPALIVE_THRESHOLD)
+            int thresh = timeouts.tcp_keep_alive_timeout.totalMilliseconds() / 2;
+            if (thresh < 5000)
+                thresh = 5000;
+            int intvl = timeouts.tcp_keep_alive_timeout.totalSeconds() / 6;
+            if (intvl < 2)
+                intvl = 2;
+            socket->setOption(IPPROTO_TCP, TCP_KEEPALIVE_THRESHOLD, thresh);
+            socket->setOption(IPPROTO_TCP, TCP_KEEPINTVL, intvl);
+            socket->setOption(IPPROTO_TCP, TCP_KEEPCNT, 3);
+#elif defined(TCP_KEEPALIVE)
+            socket->setOption(IPPROTO_TCP, TCP_KEEPALIVE, timeouts.tcp_keep_alive_timeout);
+#elif defined(TCP_KEEPIDLE)
+            socket->setOption(IPPROTO_TCP, TCP_KEEPIDLE, timeouts.tcp_keep_alive_timeout);
 #endif
-                , timeouts.tcp_keep_alive_timeout);
         }
 
         in = std::make_shared<ReadBufferFromPocoSocket>(*socket);
-- 
2.40.1

