From ff3837dec085aed076b3e338e8cda235b849c57b Mon Sep 17 00:00:00 2001
From: Alex Wilson <alex@cooperi.net>
Date: Tue, 30 May 2023 10:59:27 +1000
Subject: [PATCH 18/42] 20-preload-garbage.patch

diff -wpruN --no-dereference '--exclude=*.orig' a~/PreLoad.cmake a/PreLoad.cmake
--- a~/PreLoad.cmake	1970-01-01 00:00:00
+++ a/PreLoad.cmake	1970-01-01 00:00:00
@@ -37,17 +37,6 @@ if (NOT "$ENV{CFLAGS}" STREQUAL ""
     message("CMAKE_CXX_FLAGS_INIT: ${CMAKE_CXX_FLAGS_INIT}")
     message("CMAKE_EXE_LINKER_FLAGS_INIT: ${CMAKE_EXE_LINKER_FLAGS_INIT}")
     message("CMAKE_MODULE_LINKER_FLAGS_INIT: ${CMAKE_MODULE_LINKER_FLAGS_INIT}")
-
-    message(FATAL_ERROR "
-        Some of the variables like CFLAGS, CXXFLAGS, LDFLAGS are not empty.
-        It is not possible to build ClickHouse with custom flags.
-        These variables can be set up by previous invocation of some other build tools.
-        You should cleanup these variables and start over again.
-
-        Run the `env` command to check the details.
-        You will also need to remove the contents of the build directory.
-
-        Note: if you don't like this behavior, you can manually edit the cmake files, but please don't complain to developers.")
 endif()
 
 # Default toolchain - this is needed to avoid dependency on OS files.
