From bda731048744e14246a93dcd82edb701e86c256e Mon Sep 17 00:00:00 2001
From: Alex Wilson <alex@cooperi.net>
Date: Tue, 30 May 2023 11:40:16 +1000
Subject: [PATCH 21/42] 23-sunos-default-libs.patch

diff -wpruN --no-dereference '--exclude=*.orig' a~/cmake/sunos/default_libs.cmake a/cmake/sunos/default_libs.cmake
--- a~/cmake/sunos/default_libs.cmake	1970-01-01 00:00:00
+++ a/cmake/sunos/default_libs.cmake	1970-01-01 00:00:00
@@ -0,0 +1,30 @@
+set (DEFAULT_LIBS "-nodefaultlibs")
+
+if (NOT COMPILER_CLANG)
+    message (FATAL_ERROR "illumos build is supported only for Clang")
+endif ()
+
+set (BUILTINS_LIBRARY "-lgcc_s")
+
+set (DEFAULT_LIBS "${DEFAULT_LIBS} ${BUILTINS_LIBRARY} ${COVERAGE_OPTION} -lc -lm -lsocket -lnsl -lsendfile -lproc")
+
+message(STATUS "Default libraries: ${DEFAULT_LIBS}")
+
+set(CMAKE_CXX_STANDARD_LIBRARIES ${DEFAULT_LIBS})
+set(CMAKE_C_STANDARD_LIBRARIES ${DEFAULT_LIBS})
+
+set(CMAKE_HAVE_LIBC_PTHREAD ON)
+find_package(Threads REQUIRED)
+
+include (cmake/unwind.cmake)
+include (cmake/cxx.cmake)
+
+target_link_libraries(global-group INTERFACE
+    $<TARGET_PROPERTY:global-libs,INTERFACE_LINK_LIBRARIES>
+)
+
+# FIXME: remove when all contribs will get custom cmake lists
+install(
+    TARGETS global-group global-libs
+    EXPORT global
+)
diff -wpruN --no-dereference '--exclude=*.orig' a~/CMakeLists.txt a/CMakeLists.txt
--- a~/CMakeLists.txt	1970-01-01 00:00:00
+++ a/CMakeLists.txt	1970-01-01 00:00:00
@@ -388,6 +388,8 @@ elseif (OS_DARWIN)
     include(cmake/darwin/default_libs.cmake)
 elseif (OS_FREEBSD)
     include(cmake/freebsd/default_libs.cmake)
+elseif (OS_SUNOS)
+    include(cmake/sunos/default_libs.cmake)
 else()
     link_libraries(global-group)
 endif ()
