From a7156156f4233a0196fece849a485f3815c1d110 Mon Sep 17 00:00:00 2001
From: Alex Wilson <alex@cooperi.net>
Date: Tue, 30 May 2023 11:41:40 +1000
Subject: [PATCH 22/42] 24-sunos-locale-bits.patch

diff -wpruN --no-dereference '--exclude=*.orig' a~/contrib/llvm-project/libcxx/include/__locale a/contrib/llvm-project/libcxx/include/__locale
--- a~/contrib/llvm-project/libcxx/include/__locale	1970-01-01 00:00:00
+++ a/contrib/llvm-project/libcxx/include/__locale	1970-01-01 00:00:00
@@ -549,8 +549,10 @@ public:
 
     _LIBCPP_INLINE_VISIBILITY ctype_base() {}
 
+#ifndef __sun
     static_assert((__regex_word & ~(space | print | cntrl | upper | lower | alpha | digit | punct | xdigit | blank)) == __regex_word,
                   "__regex_word can't overlap other bits");
+#endif
 };
 
 template <class _CharT> class _LIBCPP_TEMPLATE_VIS ctype;
