From cee4c67191f2db7a5ebd2a5b6d217f5e050a4328 Mon Sep 17 00:00:00 2001
From: Alex Wilson <alex@cooperi.net>
Date: Mon, 5 Jun 2023 13:19:44 +1000
Subject: [PATCH 28/42] 30-more-cs-ds.patch

diff -wpruN --no-dereference '--exclude=*.orig' a~/contrib/boost/boost/geometry/strategies/line_interpolate/cartesian.hpp a/contrib/boost/boost/geometry/strategies/line_interpolate/cartesian.hpp
--- a~/contrib/boost/boost/geometry/strategies/line_interpolate/cartesian.hpp	1970-01-01 00:00:00
+++ a/contrib/boost/boost/geometry/strategies/line_interpolate/cartesian.hpp	1970-01-01 00:00:00
@@ -56,10 +56,10 @@ struct default_strategy<Geometry, cartes
 };
 
 
-template <typename CT, typename DS>
-struct strategy_converter<strategy::line_interpolate::cartesian<CT, DS> >
+template <typename CT, typename DST>
+struct strategy_converter<strategy::line_interpolate::cartesian<CT, DST> >
 {
-    static auto get(strategy::line_interpolate::cartesian<CT, DS> const&)
+    static auto get(strategy::line_interpolate::cartesian<CT, DST> const&)
     {
         return strategies::line_interpolate::cartesian<CT>();
     }
diff -wpruN --no-dereference '--exclude=*.orig' a~/contrib/boost/boost/geometry/strategies/line_interpolate/spherical.hpp a/contrib/boost/boost/geometry/strategies/line_interpolate/spherical.hpp
--- a~/contrib/boost/boost/geometry/strategies/line_interpolate/spherical.hpp	1970-01-01 00:00:00
+++ a/contrib/boost/boost/geometry/strategies/line_interpolate/spherical.hpp	1970-01-01 00:00:00
@@ -72,12 +72,12 @@ struct default_strategy<Geometry, spheri
 };
 
 
-template <typename CT, typename DS>
-struct strategy_converter<strategy::line_interpolate::spherical<CT, DS> >
+template <typename CT, typename DST>
+struct strategy_converter<strategy::line_interpolate::spherical<CT, DST> >
 {
-    static auto get(strategy::line_interpolate::spherical<CT, DS> const& s)
+    static auto get(strategy::line_interpolate::spherical<CT, DST> const& s)
     {
-        typedef typename strategy::line_interpolate::spherical<CT, DS>::radius_type radius_type;
+        typedef typename strategy::line_interpolate::spherical<CT, DST>::radius_type radius_type;
         return strategies::line_interpolate::spherical<radius_type, CT>(s.radius());
     }
 };
diff -wpruN --no-dereference '--exclude=*.orig' a~/contrib/llvm-project/llvm/include/llvm/IR/DebugInfoMetadata.h a/contrib/llvm-project/llvm/include/llvm/IR/DebugInfoMetadata.h
--- a~/contrib/llvm-project/llvm/include/llvm/IR/DebugInfoMetadata.h	1970-01-01 00:00:00
+++ a/contrib/llvm-project/llvm/include/llvm/IR/DebugInfoMetadata.h	1970-01-01 00:00:00
@@ -563,18 +563,18 @@ private:
   Optional<MDString *> Source;
 
   DIFile(LLVMContext &C, StorageType Storage,
-         Optional<ChecksumInfo<MDString *>> CS, Optional<MDString *> Src,
+         Optional<ChecksumInfo<MDString *>> CSI, Optional<MDString *> Src,
          ArrayRef<Metadata *> Ops);
   ~DIFile() = default;
 
   static DIFile *getImpl(LLVMContext &Context, StringRef Filename,
                          StringRef Directory,
-                         Optional<ChecksumInfo<StringRef>> CS,
+                         Optional<ChecksumInfo<StringRef>> CSI,
                          Optional<StringRef> Source, StorageType Storage,
                          bool ShouldCreate = true) {
     Optional<ChecksumInfo<MDString *>> MDChecksum;
-    if (CS)
-      MDChecksum.emplace(CS->Kind, getCanonicalMDString(Context, CS->Value));
+    if (CSI)
+      MDChecksum.emplace(CSI->Kind, getCanonicalMDString(Context, CSI->Value));
     return getImpl(
         Context, getCanonicalMDString(Context, Filename),
         getCanonicalMDString(Context, Directory), MDChecksum,
@@ -584,7 +584,7 @@ private:
   }
   static DIFile *getImpl(LLVMContext &Context, MDString *Filename,
                          MDString *Directory,
-                         Optional<ChecksumInfo<MDString *>> CS,
+                         Optional<ChecksumInfo<MDString *>> CSI,
                          Optional<MDString *> Source, StorageType Storage,
                          bool ShouldCreate = true);
 
@@ -596,14 +596,14 @@ private:
 public:
   DEFINE_MDNODE_GET(DIFile,
                     (StringRef Filename, StringRef Directory,
-                     Optional<ChecksumInfo<StringRef>> CS = None,
+                     Optional<ChecksumInfo<StringRef>> CSI = None,
                      Optional<StringRef> Source = None),
-                    (Filename, Directory, CS, Source))
+                    (Filename, Directory, CSI, Source))
   DEFINE_MDNODE_GET(DIFile,
                     (MDString * Filename, MDString *Directory,
-                     Optional<ChecksumInfo<MDString *>> CS = None,
+                     Optional<ChecksumInfo<MDString *>> CSI = None,
                      Optional<MDString *> Source = None),
-                    (Filename, Directory, CS, Source))
+                    (Filename, Directory, CSI, Source))
 
   TempDIFile clone() const { return cloneImpl(); }
 
