From ba7447bb2c511703d676b24dbf048e15880d7cab Mon Sep 17 00:00:00 2001
From: Alex Wilson <alex@cooperi.net>
Date: Mon, 5 Jun 2023 13:20:26 +1000
Subject: [PATCH 29/42] 31-pg-attribute-printf.patch

diff -wpruN --no-dereference '--exclude=*.orig' a~/contrib/libpq/include/port.h a/contrib/libpq/include/port.h
--- a~/contrib/libpq/include/port.h	1970-01-01 00:00:00
+++ a/contrib/libpq/include/port.h	1970-01-01 00:00:00
@@ -195,6 +195,9 @@ extern unsigned char pg_ascii_tolower(un
 #undef printf
 #endif
 
+#define	pg_attribute_printf
+#define	pg_attribute_printf(X,Y)
+
 extern int	pg_vsnprintf(char *str, size_t count, const char *fmt, va_list args);
 extern int	pg_snprintf(char *str, size_t count, const char *fmt,...) pg_attribute_printf(3, 4);
 extern int	pg_vsprintf(char *str, const char *fmt, va_list args);
