From 81cd48fcf702751bfe38c5a6b2a932d39ab5de37 Mon Sep 17 00:00:00 2001
From: Alex Wilson <alex@cooperi.net>
Date: Mon, 5 Jun 2023 13:20:52 +1000
Subject: [PATCH 30/42] 32-cares-need-pollh.patch

diff -wpruN --no-dereference '--exclude=*.orig' a~/src/Common/CaresPTRResolver.cpp a/src/Common/CaresPTRResolver.cpp
--- a~/src/Common/CaresPTRResolver.cpp	1970-01-01 00:00:00
+++ a/src/Common/CaresPTRResolver.cpp	1970-01-01 00:00:00
@@ -1,6 +1,7 @@
 #include "CaresPTRResolver.h"
 #include <arpa/inet.h>
 #include <sys/select.h>
+#include <sys/poll.h>
 #include <Common/Exception.h>
 #include <Common/logger_useful.h>
 #include "ares.h"
