From 2f3263810a5ec215bc498bc186636903d28f13ac Mon Sep 17 00:00:00 2001
From: Alex Wilson <alex@cooperi.net>
Date: Mon, 5 Jun 2023 13:26:08 +1000
Subject: [PATCH 33/39] 35-sunos-no-strerror_r.patch

---
 contrib/libpq/include/port.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/contrib/libpq/include/port.h b/contrib/libpq/include/port.h
index f0d4e702c2..48a61f2e40 100644
--- a/contrib/libpq/include/port.h
+++ b/contrib/libpq/include/port.h
@@ -233,7 +233,9 @@ extern char *pg_strerror(int errnum);
 
 /* Likewise for strerror_r(); note we prefer the GNU API for that */
 extern char *pg_strerror_r(int errnum, char *buf, size_t buflen);
+#if !defined(__sun)
 #define strerror_r pg_strerror_r
+#endif
 #define PG_STRERROR_R_BUFLEN 256	/* Recommended buffer size for strerror_r */
 
 /* Wrap strsignal(), or provide our own version if necessary */
-- 
2.40.1

