From 21920f173253a0a6e817db6fbdc98c561eab1486 Mon Sep 17 00:00:00 2001
From: Alex Wilson <alex@cooperi.net>
Date: Mon, 5 Jun 2023 13:26:37 +1000
Subject: [PATCH 34/42] 36-cstdlib-in-nats.patch

diff -wpruN --no-dereference '--exclude=*.orig' a~/src/Storages/NATS/NATSHandler.h a/src/Storages/NATS/NATSHandler.h
--- a~/src/Storages/NATS/NATSHandler.h	1970-01-01 00:00:00
+++ a/src/Storages/NATS/NATSHandler.h	1970-01-01 00:00:00
@@ -1,5 +1,6 @@
 #pragma once
 
+#include <cstdlib>
 #include <uv.h>
 #include <memory>
 #include <mutex>
