From 04a1262d18fab793a8769942ed8baa50850ee01a Mon Sep 17 00:00:00 2001
From: Alex Wilson <alex@cooperi.net>
Date: Mon, 5 Jun 2023 13:27:20 +1000
Subject: [PATCH 35/42] 37-rlimit_nproc-is-a-linuxism.patch

diff -wpruN --no-dereference '--exclude=*.orig' a~/programs/server/Server.cpp a/programs/server/Server.cpp
--- a~/programs/server/Server.cpp	1970-01-01 00:00:00
+++ a/programs/server/Server.cpp	1970-01-01 00:00:00
@@ -938,6 +938,7 @@ try
         }
     }
 
+#if defined(RLIMIT_NPROC)
     /// Try to increase limit on number of threads.
     {
         rlimit rlim;
@@ -969,6 +970,7 @@ try
             global_context->addWarningMessage("Maximum number of threads is lower than 30000. There could be problems with handling a lot of simultaneous queries.");
         }
     }
+#endif
 
     static ServerErrorHandler error_handler;
     Poco::ErrorHandler::set(&error_handler);
