From bfeb1841953ff4a522a63e83def890d974610328 Mon Sep 17 00:00:00 2001
From: Alex Wilson <alex@cooperi.net>
Date: Mon, 5 Jun 2023 13:35:06 +1000
Subject: [PATCH 36/42] 38-horrifying-dso-thread_local-hack.patch

diff -wpruN --no-dereference '--exclude=*.orig' a~/src/Common/ThreadStatus.cpp a/src/Common/ThreadStatus.cpp
--- a~/src/Common/ThreadStatus.cpp	1970-01-01 00:00:00
+++ a/src/Common/ThreadStatus.cpp	1970-01-01 00:00:00
@@ -10,6 +10,10 @@
 #include <csignal>
 #include <sys/mman.h>
 
+extern "C" {
+extern char __dso_handle;
+}
+
 
 namespace DB
 {
