From e128439448ca870e0f620f0b4a05d4981fa76e31 Mon Sep 17 00:00:00 2001
From: Alex Wilson <alex@cooperi.net>
Date: Wed, 7 Jun 2023 14:12:35 +1000
Subject: [PATCH 40/42] 42-backport-tls-client-auth.patch

---
 src/Server/TCPHandler.cpp | 25 +++++++++++++++++++++++++
 1 file changed, 25 insertions(+)

diff --git a/src/Server/TCPHandler.cpp b/src/Server/TCPHandler.cpp
index b268179ae9..8209efc87b 100644
--- a/src/Server/TCPHandler.cpp
+++ b/src/Server/TCPHandler.cpp
@@ -44,6 +44,8 @@
 #include <Common/logger_useful.h>
 #include <Common/CurrentMetrics.h>
 #include <fmt/format.h>
+#include <Poco/Net/SecureStreamSocket.h>
+#include <Poco/Net/SecureStreamSocketImpl.h>
 
 #include <Processors/Executors/PullingAsyncPipelineExecutor.h>
 #include <Processors/Executors/PushingPipelineExecutor.h>
@@ -1113,6 +1115,29 @@ void TCPHandler::receiveHello()
     }
 
     session = makeSession();
+
+#if USE_SSL
+    /// Authentication with SSL user certificate
+    if (dynamic_cast<Poco::Net::SecureStreamSocketImpl*>(socket().impl()))
+    {
+        Poco::Net::SecureStreamSocket secure_socket(socket());
+        if (secure_socket.havePeerCertificate())
+        {
+            try
+            {
+                session->authenticate(
+                    SSLCertificateCredentials{user, secure_socket.peerCertificate().commonName()},
+                    socket().peerAddress());
+                return;
+            }
+            catch (...)
+            {
+                tryLogCurrentException(log, "SSL authentication failed, falling back to password authentication");
+            }
+        }
+    }
+#endif
+
     session->authenticate(user, password, socket().peerAddress());
 }
 
-- 
2.40.1

