
See https://github.com/ClickHouse/ClickHouse/pull/52833

diff -wpruN --no-dereference '--exclude=*.orig' a~/src/Server/KeeperTCPHandler.cpp a/src/Server/KeeperTCPHandler.cpp
--- a~/src/Server/KeeperTCPHandler.cpp	1970-01-01 00:00:00
+++ a/src/Server/KeeperTCPHandler.cpp	1970-01-01 00:00:00
@@ -175,10 +175,12 @@ struct SocketInterruptablePollWrapper
             }
             while (rc < 0 && errno == POCO_EINTR);
 
-            if (rc >= 1 && poll_buf[0].revents & POLLIN)
+            if (rc >= 1) {
+                if (poll_buf[0].revents & POLLIN)
                 socket_ready = true;
-            if (rc >= 2 && poll_buf[1].revents & POLLIN)
+                if (poll_buf[1].revents & POLLIN)
                 fd_ready = true;
+            }
 #endif
         }
 
