From 3f5651a6db8a57acb4ae1f0b71472355165d24a8 Mon Sep 17 00:00:00 2001
From: Andy Fiddaman <illumos@fiddaman.net>
Date: Tue, 17 Oct 2023 17:04:17 +0000
Subject: [PATCH 4/5] keeperstorage

diff -wpruN --no-dereference '--exclude=*.orig' a~/src/Coordination/KeeperStorage.h a/src/Coordination/KeeperStorage.h
--- a~/src/Coordination/KeeperStorage.h	1970-01-01 00:00:00
+++ a/src/Coordination/KeeperStorage.h	1970-01-01 00:00:00
@@ -280,7 +280,7 @@ public:
 
         struct Hash
         {
-            auto operator()(const std::string_view view) const
+            size_t operator()(const std::string_view view) const
             {
                 SipHash hash;
                 hash.update(view);
