From 08598776bc35e3ae006ccff6c2a4db1254fa0f1b Mon Sep 17 00:00:00 2001
From: Andy Fiddaman <illumos@fiddaman.net>
Date: Mon, 20 Nov 2023 12:35:09 +0000
Subject: [PATCH 3/3] link libunwind

diff -wpruN --no-dereference '--exclude=*.orig' a~/cmake/sunos/default_libs.cmake a/cmake/sunos/default_libs.cmake
--- a~/cmake/sunos/default_libs.cmake	1970-01-01 00:00:00
+++ a/cmake/sunos/default_libs.cmake	1970-01-01 00:00:00
@@ -7,6 +7,8 @@ endif ()
 set (BUILTINS_LIBRARY "-lgcc_s")
 
 set (DEFAULT_LIBS "${DEFAULT_LIBS} ${BUILTINS_LIBRARY} ${COVERAGE_OPTION} -lc -lm -lsocket -lnsl -lsendfile -lproc")
+# XXX Why is this not being done by default, and where should it live?
+set (DEFAULT_LIBS "${DEFAULT_LIBS} -Lcontrib/libunwind-cmake/ -lunwind")
 
 message(STATUS "Default libraries: ${DEFAULT_LIBS}")
 
diff -wpruN --no-dereference '--exclude=*.orig' a~/CMakeLists.txt a/CMakeLists.txt
--- a~/CMakeLists.txt	1970-01-01 00:00:00
+++ a/CMakeLists.txt	1970-01-01 00:00:00
@@ -156,7 +156,7 @@ elseif(GLIBC_COMPATIBILITY)
 endif ()
 
 # Make sure the final executable has symbols exported
-set (CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -rdynamic")
+#set (CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -rdynamic")
 
 if (OS_DARWIN)
     # The `-all_load` flag forces loading of all symbols from all libraries,
