diff --git a/contrib/thrift/lib/cpp/src/thrift/transport/TSocket.cpp b/contrib/thrift/lib/cpp/src/thrift/transport/TSocket.cpp
index c6c2bfa00..ca6759b95 100644
--- a/contrib/thrift/lib/cpp/src/thrift/transport/TSocket.cpp
+++ b/contrib/thrift/lib/cpp/src/thrift/transport/TSocket.cpp
@@ -23,6 +23,7 @@
 #include <sstream>
 #ifdef HAVE_SYS_IOCTL_H
 #include <sys/ioctl.h>
+#include <sys/filio.h>
 #endif
 #ifdef HAVE_SYS_SOCKET_H
 #include <sys/socket.h>
