diff --git a/contrib/boost/boost/system/detail/std_interoperability.hpp b/contrib/boost/boost/system/detail/std_interoperability.hpp
index 5a4e7b1..96b6043 100644
--- a/contrib/boost/boost/system/detail/std_interoperability.hpp
+++ b/contrib/boost/boost/system/detail/std_interoperability.hpp
@@ -10,6 +10,7 @@
 #include <system_error>
 #include <map>
 #include <memory>
+#include <mutex>
 
 //
 
Submodule contrib/boringssl contains modified content
