diff --git a/src/Storages/PostgreSQL/PostgreSQLPoolWithFailover.h b/src/Storages/PostgreSQL/PostgreSQLPoolWithFailover.h
index 8f6027c..048200f 100644
--- a/src/Storages/PostgreSQL/PostgreSQLPoolWithFailover.h
+++ b/src/Storages/PostgreSQL/PostgreSQLPoolWithFailover.h
@@ -1,5 +1,6 @@
 #pragma once
 
+#include <mutex>
 #include <Core/Types.h>
 #include <Poco/Util/AbstractConfiguration.h>
 #include "PostgreSQLConnectionPool.h"
