diff --git a/src/AggregateFunctions/QuantileExact.h b/src/AggregateFunctions/QuantileExact.h
index aa7b4bb..7eab35a 100644
--- a/src/AggregateFunctions/QuantileExact.h
+++ b/src/AggregateFunctions/QuantileExact.h
@@ -8,7 +8,6 @@
 #include <common/sort.h>
 #include <common/types.h>
 
-
 namespace DB
 {
 
@@ -271,11 +270,11 @@ struct QuantileExactLow : public QuantileExactBase<Value, QuantileExactLow<Value
                 auto s = array.size();
                 if (s % 2 == 1)
                 {
-                    return array[static_cast<size_t>(floor(s / 2))];
+                    return array[static_cast<size_t>(static_cast<double>(floor(static_cast<double>(s / 2))))];
                 }
                 else
                 {
-                    return array[static_cast<size_t>((floor(s / 2)) - 1)];
+                    return array[static_cast<size_t>((floor(static_cast<double>(s / 2))) - 1)];
                 }
             }
             // else quantile is the nth index of the sorted array obtained by multiplying
@@ -303,11 +302,11 @@ struct QuantileExactLow : public QuantileExactBase<Value, QuantileExactLow<Value
                     auto s = array.size();
                     if (s % 2 == 1)
                     {
-                        result[indices[i]] = array[static_cast<size_t>(floor(s / 2))];
+                        result[indices[i]] = array[static_cast<size_t>(floor(static_cast<double>(s / 2)))];
                     }
                     else
                     {
-                        result[indices[i]] = array[static_cast<size_t>(floor((s / 2) - 1))];
+                        result[indices[i]] = array[static_cast<size_t>(floor(static_cast<double>((s / 2) - 1)))];
                     }
                 }
                 // else quantile is the nth index of the sorted array obtained by multiplying
@@ -343,7 +342,7 @@ struct QuantileExactHigh : public QuantileExactBase<Value, QuantileExactHigh<Val
             if (level == 0.5)
             {
                 auto s = array.size();
-                return array[static_cast<size_t>(floor(s / 2))];
+                return array[static_cast<size_t>(floor(static_cast<double>(s / 2)))];
             }
             // else quantile is the nth index of the sorted array obtained by multiplying
             // level and size of array. Example if level = 0.1 and size of array is 10.
@@ -367,7 +366,7 @@ struct QuantileExactHigh : public QuantileExactBase<Value, QuantileExactHigh<Val
                 if (level == 0.5)
                 {
                     auto s = array.size();
-                    result[indices[i]] = array[static_cast<size_t>(floor(s / 2))];
+                    result[indices[i]] = array[static_cast<size_t>(floor(static_cast<double>(s / 2)))];
                 }
                 // else quantile is the nth index of the sorted array obtained by multiplying
                 // level and size of array. Example if level = 0.1 and size of array is 10.
