diff --git a/src/Common/tests/symbol_index.cpp b/src/Common/tests/symbol_index.cpp
index 496fa7dc3f..dd86268b0e 100644
--- a/src/Common/tests/symbol_index.cpp
+++ b/src/Common/tests/symbol_index.cpp
@@ -32,7 +32,12 @@ int main(int argc, char ** argv)
         std::cout << elem.name << ": " << elem.address_begin << " ... " << elem.address_end << "\n";
     std::cout << "\n";
 
+#if defined(OS_SUNOS)
+    // SunOS's dladdr accepts a `void *` as first argument, not `const void*`.
+    void * address = reinterpret_cast<void*>(std::stoull(argv[1], nullptr, 16));
+#else
     const void * address = reinterpret_cast<void*>(std::stoull(argv[1], nullptr, 16));
+#endif
 
     const auto * symbol = symbol_index.findSymbol(address);
     if (symbol)
