diff --git a/contrib/cppkafka/include/cppkafka/detail/endianness.h b/contrib/cppkafka/include/cppkafka/detail/endianness.h
index 21594c1..b12540a 100644
--- a/contrib/cppkafka/include/cppkafka/detail/endianness.h
+++ b/contrib/cppkafka/include/cppkafka/detail/endianness.h
@@ -109,6 +109,10 @@
 #   define __LITTLE_ENDIAN LITTLE_ENDIAN
 #   define __PDP_ENDIAN    PDP_ENDIAN
 
+#elif defined(__sun)
+
+#   include <endian.h>
+
 #else
 
 #   error platform not supported
Submodule contrib/grpc contains modified content
