diff --git a/contrib/NuRaft/include/libnuraft/attr_unused.hxx b/contrib/NuRaft/include/libnuraft/attr_unused.hxx
index 80c91bb..611bb30 100644
--- a/contrib/NuRaft/include/libnuraft/attr_unused.hxx
+++ b/contrib/NuRaft/include/libnuraft/attr_unused.hxx
@@ -18,7 +18,7 @@ limitations under the License.
 #pragma once
 
 #ifndef ATTR_UNUSED
-#if defined(__linux__) || defined(__APPLE__)
+#if defined(__linux__) || defined(__APPLE__) || defined(__sun)
     #define ATTR_UNUSED __attribute__((unused))
 #elif defined(WIN32) || defined(_WIN32)
     #define ATTR_UNUSED
