diff --git a/contrib/rocksdb/monitoring/perf_context.cc b/contrib/rocksdb/monitoring/perf_context.cc
index 53f502405..07dfa2b89 100644
--- a/contrib/rocksdb/monitoring/perf_context.cc
+++ b/contrib/rocksdb/monitoring/perf_context.cc
@@ -13,7 +13,7 @@ namespace ROCKSDB_NAMESPACE {
 PerfContext perf_context;
 #else
 #if defined(OS_SOLARIS)
-__thread PerfContext perf_context_;
+thread_local PerfContext perf_context_;
 #else
 thread_local PerfContext perf_context;
 #endif
