diff --git a/src/Common/ThreadProfileEvents.h b/src/Common/ThreadProfileEvents.h
index d96339a..c43102d 100644
--- a/src/Common/ThreadProfileEvents.h
+++ b/src/Common/ThreadProfileEvents.h
@@ -105,7 +105,11 @@ struct RUsageCounters
     {
         ::rusage rusage {};
 #if !defined(__APPLE__)
+#if defined(OS_SUNOS)
+	::getrusage(RUSAGE_LWP, &rusage);
+#else
         ::getrusage(RUSAGE_THREAD, &rusage);
+#endif
 #endif
         return RUsageCounters(rusage, getClockMonotonic());
     }
