diff --git a/src/Common/OpenTelemetryTraceContext.h b/src/Common/OpenTelemetryTraceContext.h
index fe88981..c88003b 100644
--- a/src/Common/OpenTelemetryTraceContext.h
+++ b/src/Common/OpenTelemetryTraceContext.h
@@ -11,6 +11,7 @@ struct OpenTelemetryTraceContext
     // The incoming tracestate header and the trace flags, we just pass them
     // downstream. See https://www.w3.org/TR/trace-context/
     String tracestate;
+    typedef uint8_t __uint8_t;
     __uint8_t trace_flags = 0;
 
     // Parse/compose OpenTelemetry traceparent header.
