diff --git a/contrib/NuRaft/src/crc32.hxx b/contrib/NuRaft/src/crc32.hxx
index 938cf8d..b704467 100644
--- a/contrib/NuRaft/src/crc32.hxx
+++ b/contrib/NuRaft/src/crc32.hxx
@@ -22,7 +22,7 @@ limitations under the License.
 #define _JSAHN_CRC32_H
 
 #include <stdint.h>
-#if defined(__linux__) || defined(__APPLE__)
+#if defined(__linux__) || defined(__APPLE__) || defined(__sun)
     #include <unistd.h>
 #endif
 
Submodule contrib/abseil-cpp contains modified content
