diff --git a/programs/install/Install.cpp b/programs/install/Install.cpp
index ef72624..ce755c5 100644
--- a/programs/install/Install.cpp
+++ b/programs/install/Install.cpp
@@ -830,8 +830,8 @@ namespace
                 fmt::print("The pidof command returned unusual output.\n");
             }
 
-            WriteBufferFromFileDescriptor stderr(STDERR_FILENO);
-            copyData(sh->err, stderr);
+            WriteBufferFromFileDescriptor std_err(STDERR_FILENO);
+            copyData(sh->err, std_err);
 
             sh->tryWait();
         }
