diff --git a/contrib/rocksdb/util/xxh3p.h b/contrib/rocksdb/util/xxh3p.h
index 05696cecd..3a883f422 100644
--- a/contrib/rocksdb/util/xxh3p.h
+++ b/contrib/rocksdb/util/xxh3p.h
@@ -61,11 +61,15 @@
 
 /* ===   Compiler specifics   === */
 
-#if defined (__STDC_VERSION__) && __STDC_VERSION__ >= 199901L   /* >= C99 */
-#  define XXH_RESTRICT   restrict
+#if defined(__cplusplus)
+#  define XXH_RESTRICT __restrict__
 #else
+#  if defined (__STDC_VERSION__) && __STDC_VERSION__ >= 199901L   /* >= C99 */
+#    define XXH_RESTRICT   restrict
+#  else
 /* note : it might be useful to define __restrict or __restrict__ for some C++ compilers */
-#  define XXH_RESTRICT   /* disable */
+#    define XXH_RESTRICT   /* disable */
+#  endif
 #endif
 
 #if defined(__GNUC__)
Submodule contrib/thrift contains modified content
