diff --git a/src/Common/tests/int_hashes_perf.cpp b/src/Common/tests/int_hashes_perf.cpp
index 543326a9e5..8a32f8f536 100644
--- a/src/Common/tests/int_hashes_perf.cpp
+++ b/src/Common/tests/int_hashes_perf.cpp
@@ -12,7 +12,7 @@
 
 static void setAffinity()
 {
-#if !defined(__APPLE__) && !defined(__FreeBSD__)
+#if !defined(__APPLE__) && !defined(__FreeBSD__) && !defined(__sun)
     cpu_set_t mask;
     CPU_ZERO(&mask);
     CPU_SET(0, &mask);
