From 15596bcb5fd4dc4be268c1e81123d9686c121df7 Mon Sep 17 00:00:00 2001
From: Andy Fiddaman <illumos@fiddaman.net>
Date: Tue, 17 Oct 2023 16:59:13 +0000
Subject: [PATCH 3/5] statemachine

diff -wpruN --no-dereference '--exclude=*.orig' a~/src/Coordination/KeeperStateMachine.cpp a/src/Coordination/KeeperStateMachine.cpp
--- a~/src/Coordination/KeeperStateMachine.cpp	1970-01-01 00:00:00
+++ a/src/Coordination/KeeperStateMachine.cpp	1970-01-01 00:00:00
@@ -187,7 +187,7 @@ KeeperStorage::RequestForSession KeeperS
 
     if (!buffer.eof())
     {
-        request_for_session.digest.emplace();
+        request_for_session.digest = KeeperStorage::Digest();
         readIntBinary(request_for_session.digest->version, buffer);
         if (request_for_session.digest->version != KeeperStorage::DigestVersion::NO_DIGEST)
             readIntBinary(request_for_session.digest->value, buffer);
