diff --git a/src/Functions/PerformanceAdaptors.h b/src/Functions/PerformanceAdaptors.h
index 5d4d013..716bb65 100644
--- a/src/Functions/PerformanceAdaptors.h
+++ b/src/Functions/PerformanceAdaptors.h
@@ -99,7 +99,7 @@ namespace detail
             ///  when there is no statistical significant difference between them.
             double sigma() const
             {
-                return mean() / sqrt(adjustedCount());
+                return mean() / sqrt(static_cast<double>(adjustedCount()));
             }
 
             void run()
