diff --git a/src/Storages/MergeTree/SimpleMergeSelector.cpp b/src/Storages/MergeTree/SimpleMergeSelector.cpp
index 972c6ea..b39a6ec 100644
--- a/src/Storages/MergeTree/SimpleMergeSelector.cpp
+++ b/src/Storages/MergeTree/SimpleMergeSelector.cpp
@@ -205,8 +205,8 @@ SimpleMergeSelector::PartsRange SimpleMergeSelector::select(
     Estimator estimator;
 
     /// Precompute logarithm of settings boundaries, because log function is quite expensive in terms of performance
-    const double min_size_to_lower_base_log = log(1 + settings.min_size_to_lower_base);
-    const double max_size_to_lower_base_log = log(1 + settings.max_size_to_lower_base);
+    const double min_size_to_lower_base_log = log(static_cast<double>(1 + settings.min_size_to_lower_base));
+    const double max_size_to_lower_base_log = log(static_cast<double>(1 + settings.max_size_to_lower_base));
 
     for (const auto & part_range : parts_ranges)
         selectWithinPartition(part_range, max_total_size_to_merge, estimator, settings, min_size_to_lower_base_log, max_size_to_lower_base_log);
